doc/2/getting-started/flutter/index.md

```diff
@@ -0,0 +1,138 @@
+To keep it simple, this does not use the authentication system of Kuzzle but feel free to implement it for a better usage.
+
+_login.dart:_
+
+<<< ./snippets/login.dart.snippet[dart]
+
+## The chat
+
+We have to connect the server so that our client can interact with it (note that the host is the API of an android emulator in this example).
+
+In our _chat.dart_ let's import the sdk:
+
+<<< ./snippets/chat.dart.snippet:1[dart]
+
+Then we will [establish the connection](/sdk/dart/2/core-classes/kuzzle/connect) to kuzzle and create, if they don't [exist](/sdk/dart/2/controllers/index/exists/), the [index](/sdk/dart/2/controllers/index/create) and [collection](/sdk/dart/2/controllers/collection/create) of our chat. We will also fetch messages sorted by creation date, and  subscribe to the same collection to receive new messages in realtime.
+
+### Create index and collection
+
+So first let's write a method which will create the index/collection if it does not exist:
+
+<<< ./snippets/chat.dart.snippet:2[dart]
+
+
+### Get existing messages
+
+The method _fetchMessage()_ will [search](/sdk/dart/2/controllers/document/search) for the first hundred newest messages.
+It will then update the state of the widget to store those messages in the `messages` variable.
+
+<<< ./snippets/chat.dart.snippet:3[dart]
+
+### Receive new messages in realtime
+
+And finally the method to subscribe to our collection.
+It will call the Kuzzle's realtime controller to allow us to receive [real-time notifications](/core/2/api/essentials/notifications) on message creations. New messages received that way will then be added to our list of previously fetched `messages`, and rendered in our list, by updating the state.
+
+<<< ./snippets/chat.dart.snippet:4[dart]
+
+### Call previous written methods
+
+Then we are going to call all of those methods in the `initState` method of the State of our `StatefulWidget`:
+
+<<< ./snippets/chat.dart.snippet:5[dart]
+
+As you can see we use a model to retrieve each message from the JSON Response to a Dart class.
+
+Here:
+
+<<< ./snippets/chat.dart.snippet:6[dart]
+
+And here:
+
+<<< ./snippets/chat.dart.snippet:7[dart]
+
+So here is the Dart class used for this model
+
+_message.dart:_
+
+<<< ./snippets/message.dart.snippet[dart]
+
+### Display everything
+
+Finally let's make a view to display everything and add an input to be able to send a message in our `build()` method:
+
+_chat.dart:_
+
+<<< ./snippets/chat.dart.snippet:8[dart]
+
+And _chat_view.dart:_
+
+<<< ./snippets/chat.dart.snippet:8[dart]
+
+### Sending messages
+
+Finally let's see how to send a new message using the Kuzzle sdk:
+
+_chat.dart:_
+
+<<< ./snippets/chat.dart.snippet:9[dart]
+
+For this we simply [create](/sdk/dart/2/controllers/document/create) a document in our `messages` collection with a `username` and a `value`. Once this document is created it will trigger a notification to all clients which subscribed to this collection and receive the message.
```

Comment:
```suggestion
For this we simply [create](/sdk/dart/2/controllers/document/create) a document in our `messages` collection with a `username` and a `value`. Once this document is created it will trigger a notification to all clients who subscribed to this collection and receive the message.
```